corehq/apps/api/domain_metadata.py

```diff
@@ -62,7 +76,16 @@ def obj_get_list(self, bundle, **kwargs):
         if kwargs.get('domain'):
             return [self.obj_get(bundle, **kwargs)]
         else:
-            return list(Domain.get_all())
+            filters = {}
+            if hasattr(bundle.request, 'GET'):
+                filters = bundle.request.GET
+            domains = list(Domain.get_all())
```

Comment:
Also, the full domain documents will be in elasticsearch basically unchanged—so you can use them directly instead of also hitting couch